oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/token/TokenExpirationValidator.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.security.oauth2.server.authorization.token;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.security.oauth2.core.AbstractOAuth2Token;
+import org.springframework.security.oauth2.core.OAuth2Error;
+import org.springframework.security.oauth2.core.OAuth2ErrorCodes;
+import org.springframework.security.oauth2.core.OAuth2TokenValidator;
+import org.springframework.security.oauth2.core.OAuth2TokenValidatorResult;
+import org.springframework.util.Assert;
+
+import java.time.Instant;
+
+/**
+ * An implementation of {@link OAuth2TokenValidator} to verify a token is not expired.
+ *
+ * @author Gerardo Roza
+ * @since 0.0.4
+ */
+public class TokenExpirationValidator implements OAuth2TokenValidator<AbstractOAuth2Token> {
```

Comment:
Please remove this and instead use `JwtTimestampValidator`

Reply:
I am using the `JwtTimestampValidator` when we can decode the token as a `Jwt` @jgrandja , but we also need to cover validating non-jwt-formatted tokens, for example, to validate a `refresh_token`.
Let me know if there is any plan for the future that might affect this (maybe issuing `Jwt` for all type of tokens including `refresh_token`s, or in the opposite direction, maybe we want to support issuing just opaque tokens, which would reinfoce the need of this validator)

Reply:
`TokenExpirationValidator` is not needed. Please see [comment](https://github.com/spring-projects-experimental/spring-authorization-server/pull/161#discussion_r578581645)

Reply:
Just as an FYI, I'd prefer not to expose a public API if not needed. As soon as we expose, then we need to support, which complicates things if we need to deprecate because of later improvements. So for initial implementations, we expose the minimal set of API's and only "open it up" when it makes sense.

Reply:
That makes sense, this is now fixed